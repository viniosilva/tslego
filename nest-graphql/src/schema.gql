# ------------------------------------------------------
# THIS FILE WAS AUTOMATICALLY GENERATED (DO NOT MODIFY)
# ------------------------------------------------------

type Mutation {
  createUser(input: UserInput!): User!
  removeUserById(id: Int!): Boolean!
  updateUserById(id: Int!, input: UserInput!): User!
}

type Query {
  getUserById(id: Int!): User!
  getUsers: [User!]!
}

"""User object"""
type User {
  """The ID of the user"""
  id: Int!

  """The name of the user"""
  name: String!
}

"""User input"""
input UserInput {
  """The name of the user"""
  name: String!
}
