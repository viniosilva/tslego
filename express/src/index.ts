import { apiConfig } from './common/config';
import { logger } from './common/logger';
import { Server } from './server';
import { userController } from './user';

const routes = [userController.router];
const server = new Server(
  apiConfig.host,
  apiConfig.port,
  logger,
  apiConfig.swaggerConfig,
  routes,
);

server.start();
