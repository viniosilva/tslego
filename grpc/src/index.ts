import { apiConfig } from './common/config';
import { logger } from './common/logger';
import { Server } from './Server';
import { userHandler } from './user';

const handlers = [userHandler];

const server = new Server(apiConfig.host, apiConfig.port, handlers, logger);
server.start();
