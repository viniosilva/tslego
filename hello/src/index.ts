class Hello {
  getMessage(): string {
    return 'Hello world! (:';
  }
}

export const hello = new Hello();

console.log(hello.getMessage());
