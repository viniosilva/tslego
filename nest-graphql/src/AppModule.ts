import { Module } from '@nestjs/common';
import { GraphQLModule } from '@nestjs/graphql';
import { LoggerModule } from 'nestjs-pino';
import { httpLogger } from './common/httpLogger';
import { UserModule } from './user/UserModule';

@Module({
  imports: [
    LoggerModule.forRoot({ pinoHttp: httpLogger }),
    UserModule,
    GraphQLModule.forRoot({
      autoSchemaFile: 'src/schema.gql',
      sortSchema: true,
    }),
  ],
})
export class AppModule {}
