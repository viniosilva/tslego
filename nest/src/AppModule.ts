import { Module } from '@nestjs/common';
import { LoggerModule } from 'nestjs-pino';
import { httpLogger } from './common/httpLogger';
import { UserModule } from './user/UserModule';

@Module({
  imports: [LoggerModule.forRoot({ pinoHttp: httpLogger }), UserModule],
})
export class AppModule {}
