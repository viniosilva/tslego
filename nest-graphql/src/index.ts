import { ValidationPipe } from '@nestjs/common';
import { NestFactory } from '@nestjs/core';
import helmet from 'helmet';
import { AppModule } from './AppModule';
import { apiConfig } from './common/config';
import { logger } from './common/logger';

async function bootstrap() {
  const app = await NestFactory.create(AppModule, { logger: false });
  try {
    app.useGlobalPipes(new ValidationPipe({ transform: true }));
  } catch (err) {
    logger.error(err);
  }
  // app.use(helmet());

  await app.listen(apiConfig.port);

  logger.info(`App listening at [${apiConfig.host}:${apiConfig.port}]`);
}
bootstrap();
