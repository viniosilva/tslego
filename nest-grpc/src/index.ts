import { NestFactory } from '@nestjs/core';
import { MicroserviceOptions, Transport } from '@nestjs/microservices';
import { AppModule } from './AppModule';
import { apiConfig } from './common/config';
import { logger } from './common/logger';

async function bootstrap() {
  const url = `${apiConfig.host}:${apiConfig.port}`;
  const app = await NestFactory.createMicroservice<MicroserviceOptions>(
    AppModule,
    {
      logger: false,
      transport: Transport.GRPC,
      options: {
        package: 'user',
        protoPath: 'protos/user.proto',
        url,
      },
    },
  );

  app.listen(() => logger.info(`App listening at [${url}]`));
}

bootstrap();
