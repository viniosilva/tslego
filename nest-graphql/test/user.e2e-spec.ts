import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication, ValidationPipe } from '@nestjs/common';
import request, { Response } from 'supertest';
import { AppModule } from '../src/AppModule';

describe('AppController (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    app.useGlobalPipes(new ValidationPipe({ transform: true }));
    await app.init();
  });

  describe('getUsers', () => {
    it('should return empty user list', () => {
      const query = 'query { getUsers { id, name } }';

      request(app.getHttpServer())
        .post('/graphql')
        .send({ query })
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });

    it('should return user list', async () => {
      let query = 'mutation { createUser(input: { name: "TEST" }) { id } }';
      await request(app.getHttpServer()).post('/graphql').send({ query });

      query = 'query { getUsers { id, name } }';
      return request(app.getHttpServer())
        .post('/graphql')
        .send({ query })
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });
  });

  describe('getUserById', () => {
    it('should return user', async () => {
      let query = 'mutation { createUser(input: { name: "TEST" }) { id } }';
      await request(app.getHttpServer()).post('/graphql').send({ query });

      query = 'query { getUserById(id: 1) { id, name } }';
      return request(app.getHttpServer())
        .post('/graphql')
        .send({ query })
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });

    it('should throw not found exception', () => {
      const query = 'query { getUserById(id: 0) { id, name } }';
      return request(app.getHttpServer())
        .post('/graphql')
        .send({ query })
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });
  });

  describe('createUser', () => {
    it('should return new user', () => {
      const query =
        'mutation { createUser(input: { name: "TEST" }) { id, name } }';
      return request(app.getHttpServer())
        .post('/graphql')
        .send({ query })
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });

    it('should throw bad request exception', () => {
      const query =
        'mutation { createUser(input: { name: 123 }) { id, name } }';
      return request(app.getHttpServer())
        .post('/graphql')
        .send({ query })
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });
  });

  describe('updateUserById', () => {
    it('should return updated user', async () => {
      let query = 'mutation { createUser(input: { name: "TEST" }) { id } }';
      await request(app.getHttpServer()).post('/graphql').send({ query });

      query =
        'mutation { updateUserById(id: 1, input: { name: "TEST UPDATE" }) { id, name } }';
      return request(app.getHttpServer())
        .post('/graphql')
        .send({ query })
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });

    it('should throw bad request exception', () => {
      const query =
        'mutation { updateUserById(id: 1, input: { name: 123 }) { id, name } }';
      return request(app.getHttpServer())
        .post('/graphql')
        .send({ query })
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });

    it('should throw not found exception', () => {
      const query =
        'mutation { updateUserById(id: 0, input: { name: "TEST UPDATE" }) { id, name } }';
      return request(app.getHttpServer())
        .post('/graphql')
        .send({ query })
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });
  });

  describe('removeUserById', () => {
    it('should remove user', async () => {
      let query = 'mutation { createUser(input: { name: "TEST" }) { id } }';
      await request(app.getHttpServer()).post('/graphql').send({ query });

      query = 'mutation { removeUserById(id: 1) }';
      return request(app.getHttpServer())
        .post('/graphql')
        .send({ query })
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });
  });
});
