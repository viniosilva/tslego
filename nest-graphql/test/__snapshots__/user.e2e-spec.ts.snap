// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppController (e2e) createUser should return new user 1`] = `
Object {
  "data": Object {
    "createUser": Object {
      "id": 1,
      "name": "TEST",
    },
  },
}
`;

exports[`AppController (e2e) createUser should throw bad request exception 1`] = `
Object {
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "GRAPHQL_VALIDATION_FAILED",
      },
      "locations": Array [
        Object {
          "column": 38,
          "line": 1,
        },
      ],
      "message": "String cannot represent a non string value: 123",
    },
  ],
}
`;

exports[`AppController (e2e) getUserById should return user 1`] = `
Object {
  "data": Object {
    "getUserById": Object {
      "id": 1,
      "name": "TEST",
    },
  },
}
`;

exports[`AppController (e2e) getUserById should throw not found exception 1`] = `
Object {
  "data": null,
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "INTERNAL_SERVER_ERROR",
      },
      "locations": Array [
        Object {
          "column": 9,
          "line": 1,
        },
      ],
      "message": "User not found",
      "path": Array [
        "getUserById",
      ],
    },
  ],
}
`;

exports[`AppController (e2e) getUsers should return user list 1`] = `
Object {
  "data": Object {
    "getUsers": Array [
      Object {
        "id": 1,
        "name": "TEST",
      },
    ],
  },
}
`;

exports[`AppController (e2e) removeUserById should remove user 1`] = `
Object {
  "data": Object {
    "removeUserById": true,
  },
}
`;

exports[`AppController (e2e) updateUserById should return updated user 1`] = `
Object {
  "data": Object {
    "updateUserById": Object {
      "id": 1,
      "name": "TEST UPDATE",
    },
  },
}
`;

exports[`AppController (e2e) updateUserById should throw bad request exception 1`] = `
Object {
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "GRAPHQL_VALIDATION_FAILED",
      },
      "locations": Array [
        Object {
          "column": 49,
          "line": 1,
        },
      ],
      "message": "String cannot represent a non string value: 123",
    },
  ],
}
`;

exports[`AppController (e2e) updateUserById should throw not found exception 1`] = `
Object {
  "data": null,
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "INTERNAL_SERVER_ERROR",
      },
      "locations": Array [
        Object {
          "column": 12,
          "line": 1,
        },
      ],
      "message": "User not found",
      "path": Array [
        "updateUserById",
      ],
    },
  ],
}
`;
